hazelcast/src/main/java/com/hazelcast/internal/serialization/SerializationServiceBuilder.java

```diff
@@ -63,5 +64,7 @@
 
     SerializationServiceBuilder setInitialOutputBufferSize(int initialOutputBufferSize);
 
+    SerializationServiceBuilder setClusterService(ClusterService clusterService);
```

Comment:
Sure, I understand this.
I need the version though. Two options I see is to change the service's APIe to pass the version on the serialization call (it seems ugly). 
The other option is to have some coupling, we can introduce a new interface (a sub-interface of the ClusterService).
A listener sounds good too - don't know how feasible that is.